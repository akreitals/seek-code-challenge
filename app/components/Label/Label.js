import { width } from 'styled-system';
import Text from '../Text/Text';
import theme from '../theme';

const Label = Text.withComponent('label').extend`
    display: block;
    font-size: ${theme.fontSizes[2]}px;
    line-height: ${theme.fontSizes[5]}px;
    transform: translateY(.41em);
    font-weight: ${theme.bold};
`;

Label.displayName = 'Label';

Label.propTypes = {
    ...width.PropTypes
};

Label.defaultProps = {
    theme,
    f: 2,
    w: 1
};

export default Label;
