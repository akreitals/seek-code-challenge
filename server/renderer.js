import React from 'react';
import ReactDomServer from 'react-dom/server';
import { ServerStyleSheet } from 'styled-components';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { configureStore } from '../app/state/configureStore';
import ThemeProvider from '../app/components/ThemeProvider/ThemeProvider';

import RoutesContainer from '../app/containers/Routes/RoutesContainer';

const isProduction = process.env.NODE_ENV === 'production' || false;

const render = (html, styles) => {
    const assetsManifest =
        process.env.webpackAssets && JSON.parse(process.env.webpackAssets);

    return `
    <!DOCTYPE html>
    <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
            <meta http-equiv="x-ua-compatible" content="ie=edge">
            <title>Seek Coding Challenge</title>
            ${styles}
        </head>
        <body>
            <div id="root">${html}</div>
            <script src='${
                isProduction ? assetsManifest['/vendor.js'] : 'js/vendor.js'
            }'></script>
            <script src='${
                isProduction ? assetsManifest['/app.js'] : 'js/app.js'
            }'></script>
        </body>
    </html>
    `;
};

const serverSideRender = (req, res) => {
    const context = {};
    const sheet = new ServerStyleSheet();
    const store = configureStore();

    const appWithRouter = (
        <Provider store={store}>
            <ThemeProvider>
                <StaticRouter location={req.url} context={context}>
                    <RoutesContainer />
                </StaticRouter>
            </ThemeProvider>
        </Provider>
    );

    if (context.url) {
        return res.redirect(context.url);
    }

    const html = ReactDomServer.renderToString(
        sheet.collectStyles(appWithRouter)
    );

    const styles = sheet.getStyleTags();

    return res.status(200).send(render(html, styles));
};

export default serverSideRender;
