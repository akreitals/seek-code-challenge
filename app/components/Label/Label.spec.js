import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

describe('Label', () => {
    test('should render a <Label /> component', () => {
        const json = renderer.create(<Label />).toJSON();
        expect(json).toMatchSnapshot();
    });
});
