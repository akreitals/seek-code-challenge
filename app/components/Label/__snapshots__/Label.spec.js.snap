// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render a <Label /> component 1`] = `
<label
  className="sc-ifAKCX cqHdkM"
/>
`;
